server.c

```diff
@@ -182,31 +204,69 @@ void *client_thread(void* client_fd) {
 int main(int argc, char *argv[]) {
 
     if (argc < 2) {
-        printf("Usage: %s port [seed]\n", argv[0]);
+        print_usage(argv[0]);
         return 1;
     }
-    
-    int seed = 0;
-    if (argc == 3) {
+
+    struct options opts;
+    opts = default_options;
+
+    int port = atoi(argv[1]);
+    int c;
+    opterr = 0;
+    while ((c = getopt(argc, argv, "s:a:n:l:")) != -1) {
+        switch (c) {
         char *end;
-        seed = strtol(argv[2], &end, 10);
-        if (end == argv[2]) {
-            fprintf(stderr, "Invalid seed: %s\n", argv[2]);
+            case 's':
+                opts.random_seed = (int) strtol(optarg, &end, 10);
+                LOG("seed is %d\n", opts.random_seed);
+                if (end == optarg) {
+                    return 1;
+                }
+                break;
+            case 'a':
+                opts.adj_file = malloc(strlen(optarg) + 1);
+                if(opts.adj_file == NULL){
+                    perror("malloc");
+                    return false;
+                }
+                opts.adj_free = true;
+                strcpy(opts.adj_file, optarg);
```

Comment:
For these options (a, n, l) we can simply point at the existing strings in `argv[]` -- they are already allocated when the program starts, so no need to copy them. This would also simplify our code so we don't need the "free" flags or free steps at the end of main.

Reply:
made the changes and pushed to own fork, do I need to create a different PR?